pkgs/servers/sip/kamailio/default.nix

```diff
@@ -1,76 +1,117 @@
 { callPackage
 , fetchurl
 , lib
-, pkgs
 , stdenv
+, pkg-config
+, which
+, bison
+, flex
+, json_c
+, libevent
+, libxml2
+, mariadb-connector-c
+, pcre
+, gnugrep
+, gawk
+, coreutils
+, gdb
+, gnused
+, openssl
+
+# Extra modules to build
+, extraModules ? []
+
+# Extra buildInputs need for extraModules
+, extraBuildInputs ? []
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "kamailio";
-  version = "5.7.2";
-
-  meta = {
-    description = "Fast and flexible SIP server, proxy, SBC, and load balancer";
-    homepage = "https://www.kamailio.org/";
-    license = lib.licenses.gpl2Only;
-    maintainers = with lib.maintainers; [ mawis ];
-    platforms = lib.platforms.linux;
-  };
+  version = "5.7.3";
 
   src = fetchurl {
-    url = "https://www.kamailio.org/pub/kamailio/5.7.2/src/kamailio-${finalAttrs.version}_src.tar.gz";
-    hash = "sha256-csmgZ9qNb6kg03N9mM1/ZsMh+Ay+EHbi1aOStCJQMSI=";
+    url = "https://www.kamailio.org/pub/kamailio/${finalAttrs.version}/src/kamailio-${finalAttrs.version}_src.tar.gz";
+    hash = "sha256-x6YgsDl05OBNQZ4Iancf0Leo4mnz1pwZocZghaSY/Yw=";
   };
 
-  buildInputs = with pkgs; [
-    bison
-    flex
-    gnugrep
-    json_c.dev
-    libevent.dev
-    libxml2.dev
-    mariadb-connector-c.dev
-    pcre.dev
-  ];
+  buildInputs = [
+    json_c
+    libevent
+    libxml2
+    mariadb-connector-c
+    pcre
+    openssl
+  ] ++ extraBuildInputs ;
 
-  nativeBuildInputs = with pkgs; [
+  nativeBuildInputs = [
     pkg-config
     which
+    bison
+    flex
   ];
 
-  configurePhase = ''
+  configurePhase = let
+    modules = [
```

Comment:
How about using a parameter `modules` and set this list as default value ?

Reply:
Setting it as an input parameter has one big disadvantage: if you want to just add one module you automatically clear the default list and you have to repeat (at least most of) it, which makes it much less user friendly.

I changed to an attribute of the derivation. Now it can be easily customized with `overrideAttrs`.

Reply:
Even greater !